src/oscar/apps/dashboard/catalogue/views.py

```diff
@@ -61,7 +61,7 @@ def filter_products(queryset, user):
     return queryset.filter(stockrecords__partner__users__pk=user.pk).distinct()
 
 
-class ProductListView(SingleTableMixin, generic.TemplateView):
+class ProductListView(SingleTableMixin, generic.ListView):
```

Comment:
Needs to return data, otherwise exception rises - https://github.com/bradleyayers/django-tables2/blob/46e4e7c85f7ba230c53a6421f64c09fb1241369d/django_tables2/tables.py#L43
Otherwise, we can override `get_table_data`. WDYT?